avocado/core/job.py

```diff
@@ -533,7 +516,12 @@ def run(self):
         """
         runtime.CURRENT_JOB = self
         try:
-            return self._run()
+            self._setup_job_results()
+            self.__start_job_logging()
+
+            self.create_test_suite()
+            self.pre_tests()
+            return self.run_tests()
```

Comment:
What a nice cleanup. Btw how about moving `_setup_job_results` and `__start_job_logging` into the `pre_tests`? Or do you intend to make them public? In my eyes they are priority-0 pre-test methods, so I'd hardcode them there. (the same for `sto_job_logging` in `post`).